### PR TITLE
Feature/quaK

### PR DESCRIPTION
**Pull request by Joelrau**

### CODE REVIEW

File: src/client/component/logfile.cpp [hunk 1]
```diff
@@ -134,31 +134,47 @@ namespace logfile
...
 			{
-				hidden = true;
-				++text;
-			}
+				auto hidden = false;
+				std::string message(game::ConcatArgs(1));
 
-			const std::string message = text;
-			const auto client = ent->s.entityNum;
+				hidden = message[1] == '/';
```
Comment (momo5502) [inline src/client/component/logfile.cpp:149]: why 1 here?

File: src/client/component/network.cpp [hunk 2]
```diff
@@ -260,7 +260,7 @@ namespace network
 				on("print", [](const game::netadr_s& addr, const std::string_view& data)
 				{
 					const std::string message{data};
-					console::info("%s\n", message.data());
+					console::info(message.data());
```
Comment (momo5502) [inline src/client/component/network.cpp:263]: nah, it needs the newline, console::info/error/warn, don't add these automatically anymore, i wanted it to behave just like printf

Comment (momo5502) [reply src/client/component/network.cpp:263]: nvm, i see what you did there, it's ofc totally fine. ignore my comment :)

File: src/client/component/rcon.cpp [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+#include "command.hpp"
+#include "console.hpp"
+#include "network.hpp"
+#include "scheduler.hpp"
+
+namespace rcon
+{
+	namespace
+	{
+		bool is_redirecting_ = false;
```
Comment (momo5502) [inline src/client/component/rcon.cpp:16]: I would have made this volatile, but considering volatile is deprecated now ¯\_(ツ)_/¯

File: src/client/component/rcon.cpp [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+				}
+			}
+
+			return buffer;
+		}
+	}
+
+	bool message_redirect(const std::string& message)
+	{
+		std::lock_guard<std::recursive_mutex> $(redirect_lock);
```
Comment (momo5502) [inline src/client/component/rcon.cpp:108]: Here, what you can do is add a if(!is_redirecting_) return false; in front. That way, if redirecting is not enabled, you're not unnecessarily acquiring the mutex. And when it is, you acquire the mutex and do the stuff below as it is

**Review by momo5502 — CHANGES_REQUESTED** [review]
Small comments

**Review by momo5502 — APPROVED** [review]

File: src/client/component/network.cpp (hunk 2 above)
Comment (diamante0018) [reply src/client/component/network.cpp:263]: Oops

**Pull request merged**